examples/non-fungible/src/lib.rs

```diff
@@ -0,0 +1,147 @@
+}
+
+impl ServiceAbi for NonFungibleTokenAbi {
+    type Query = Request;
+    type QueryResponse = Response;
+    type Parameters = ();
+}
+
+/// An operation.
+#[derive(Debug, Deserialize, Serialize, GraphQLMutationRoot)]
+pub enum Operation {
+    /// Mints a token
+    Mint { name: String, payload: Vec<u8> },
+    /// Transfers a token from a (locally owned) account to a (possibly remote) account.
+    Transfer {
+        source_owner: AccountOwner,
+        token_id: TokenId,
+        target_account: Account,
+    },
+    /// Same as `Transfer` but the source account may be remote. Depending on its
+    /// configuration, the target chain may take time or refuse to process
+    /// the message.
+    Claim {
+        source_account: Account,
+        token_id: TokenId,
+        target_account: Account,
+    },
+}
+
+/// A message.
+#[derive(Debug, Deserialize, Serialize)]
+pub enum Message {
+    /// Transfers to the given `target` account, unless the message is bouncing, in which case
+    /// we transfer back to the `source`.
+    Transfer { nft: Nft, target_account: Account },
+
+    /// Claims from the given account and starts a transfer to the target account.
+    Claim {
+        source_account: Account,
+        token_id: TokenId,
+        target_account: Account,
+    },
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, SimpleObject, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct Nft {
+    pub token_id: TokenId,
+    pub owner: AccountOwner,
+    pub name: String,
+    pub minter: AccountOwner,
+    pub payload: Vec<u8>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, SimpleObject, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct NftOutput {
+    pub token_id: String,
+    pub owner: AccountOwner,
+    pub name: String,
+    pub minter: AccountOwner,
+    pub payload: Vec<u8>,
+}
+
+impl NftOutput {
+    pub fn new(nft: Nft) -> Self {
+        use base64::engine::{general_purpose::STANDARD_NO_PAD, Engine as _};
+        let token_id = STANDARD_NO_PAD.encode(nft.token_id.id);
+        Self {
+            token_id,
+            owner: nft.owner,
+            name: nft.name,
+            minter: nft.minter,
+            payload: nft.payload,
+        }
+    }
+
+    pub fn new_with_token_id(token_id: String, nft: Nft) -> Self {
+        Self {
+            token_id,
+            owner: nft.owner,
+            name: nft.name,
+            minter: nft.minter,
+            payload: nft.payload,
+        }
+    }
+}
+
+impl Display for TokenId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self.id)
+    }
+}
+
+impl Nft {
+    pub fn create_token_id(
+        chain_id: &ChainId,
+        application_id: &ApplicationId,
+        name: &String,
+        minter: &AccountOwner,
+        payload: &Vec<u8>,
+    ) -> TokenId {
+        use sha3::Digest as _;
+
+        let mut hasher = sha3::Sha3_256::new();
+        hasher.update(chain_id.to_string());
```

Comment:
(Why not just use the bytes directly for chain ID, application ID and minter, rather than turning it into a string? Is this to make it easier to compute e.g. in JavaScript?)